flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/InternalTimerHeap.java

```diff
@@ -224,6 +227,15 @@ void bulkAddRestoredTimers(Collection<? extends InternalTimer<K, N>> restoredTim
 		return result;
 	}
 
+	@Nonnull
+	StateSnapshot snapshot(TimerHeapInternalTimer.TimerSerializer<K, N> serializer) {
+		return new InternalTimerHeapSnapshot<>(
+			Arrays.copyOfRange(queue, 1, size + 1),
+			serializer,
```

Comment:
The `serializer` was be duplicated here or in the constructor of `InternalTimerHeapSnapshot` in the current code, do you plan to duplicate it when calling the `snapshot()` in the next code?

Reply:
This method is not called and the signature will probably change. Just have it here to roughly outline how it this could be used in the future.